src/DynamoCore/Nodes/Custom Nodes/Function.cs

```diff
@@ -82,6 +82,62 @@ protected override void SerializeCore(XmlElement element, SaveContext context)
             element.AppendChild(outEl);
         }
 
+        /// <summary>
+        ///     Complete a definition for a proxy custom node instance 
+        ///     by adding input and output ports as far as we don't have
+        ///     a corresponding custom node workspace
+        /// </summary>
+        /// <param name="funcID">Identifier of the custom node instance</param>
+        /// <param name="inputs">Number of inputs</param>
+        /// <param name="outputs">Number of outputs</param>
+        internal void LoadNode(Guid nodeId, int inputs, int outputs)
+        {
+            GUID = nodeId;
+            
+            // make the custom node instance be in sync 
+            // with its definition if it's needed
+            if (!Controller.IsInSyncWithNode(this))
+            {
+                Controller.SyncNodeWithDefinition(this);
+            }
+            else
+            {
+                PortData data;
+                if (outputs > -1)
```

Comment:
Please make this `if (outputs > 0)` for it to make more sense (`0` will not execute the `for` loop anyway). Do the same to `input` below.